### DISCUSSION TITLE
Read Lists: In release date ordering, if a release date clashes with another, fallback to title sort

### DISCUSSION THREAD

**Discussion by TBhimdi** [gotson/komga#1658] [Ideas]
When creating Read Lists with titles with oneshots, annuals, spin-offs so that you are able to read them in the release date order, the ordering of the issues can be swapped/incorrect.

For example, when release dates are pulled from Comic Vine and stored within the cbz, some release dates can have "fuzzy dates", e.g., "Q4 2000", "October 2008". The dates are correct, but either the exact date of the month is unknown or multiple issues were released on the same day. In those situations, where multiple consecutive issues have fuzzy dates which are the same, the Read List can swap the order of the issue numbers, so #4 is listed before #3, thus ordering is incorrect. Falling back to title sort in that situation would keep the correct reading order intact.

Some titles I'm having this issue with are Miracleman, Invincible, The Flash, and Animal Man.

Enhancement: When release dates are the same, fall back to title sort for those issues.

**Comment by gotson** [comment]
This is a metadata issue, fixing the metadata to get correct dates will fix this.

**Reply by TBhimdi** [reply]
The issue occurs when two consecutive issues are released on the same day as well. There is no metadata fix for that.

**Reply by gotson** [reply]
sorting by title will not solve this either

**Reply by TBhimdi** [reply]
Why would sorting alphabetically by title not solve that problem? Issue XXX 003 would sort before XXX 004 (right now they're swapped). Otherwise if not title, then number or sort number fields would work, for the same title anyway.

**Reply by gotson** [reply]
> Why would sorting alphabetically by title not solve that problem? Issue XXX 003 would sort before XXX 004 (right now they're swapped). Otherwise if not title, then number or sort number fields would work, for the same title anyway.

you assume that all books that have the same realease date would be ordered by alphabetical order, while it may not always be true.

Number and number sort only make sense within a series. They don't make sense in a read list, which can have books from multiple series, or one shots.

**Reply by TBhimdi** [reply]
Right, but currently in a Read List with release date ordering, if the dates match, isn't the ordering of those issues random? Having it be deterministic would be nice, it would at least fix it for same-series date clashes.

**Reply by gotson** [reply]
> Having it be deterministic would be nice, it would at least fix it for same-series date clashes.

how would it fix clashes if the dates are still the same ? There will always be one before the other, but it may or may not be accurate.

**Reply by TBhimdi** [reply]
I meant sorting by title or number if there was a clash for same-series sorting. Two consecutive issues of the same series in a Read List, ordered by release date, both issues have the same release date, then sort by title (or issue number). Right now those two could be in any order.

**Reply by gotson** [reply]
> I meant sorting by title or number if there was a clash for same-series sorting. Two consecutive issues of the same series in a Read List, ordered by release date, both issues have the same release date, then sort by title (or issue number). Right now those two could be in any order.

that's a lot of edge cases tbh, and it could still be wrong when sorting by title.
